src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.EvpPKey.MLDsa.cs

```diff
@@ -0,0 +1,179 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using Microsoft.Win32.SafeHandles;
+
+internal static partial class Interop
+{
+    internal static partial class Crypto
+    {
+        [Experimental(Experimentals.PostQuantumCryptographyDiagId)]
+        internal static partial class EvpPKeyMLDsa
+        {
+            internal static string? MLDsa44 { get; }
+            internal static string? MLDsa65 { get; }
+            internal static string? MLDsa87 { get; }
+
+            static EvpPKeyMLDsa()
+            {
+                CryptoInitializer.Initialize();
+
+                // Do not use property initializers for these because we need to ensure CryptoInitializer.Initialize
+                // is called first. Property initializers happen before cctors, so instead set the property after the
+                // initializer is run.
+                MLDsa44 = IsSignatureAlgorithmAvailable(MLDsaAlgorithm.MLDsa44.Name);
+                MLDsa65 = IsSignatureAlgorithmAvailable(MLDsaAlgorithm.MLDsa65.Name);
+                MLDsa87 = IsSignatureAlgorithmAvailable(MLDsaAlgorithm.MLDsa87.Name);
+            }
+
+            [LibraryImport(Libraries.CryptoNative, StringMarshalling = StringMarshalling.Utf8)]
+            private static partial int CryptoNative_IsSignatureAlgorithmAvailable(string algorithm);
+
+            private static string? IsSignatureAlgorithmAvailable(string algorithm)
+            {
+                const int Available = 1;
+                const int NotAvailable = 0;
+
+                int ret = CryptoNative_IsSignatureAlgorithmAvailable(algorithm);
+                return ret switch
+                {
+                    Available => algorithm,
+                    NotAvailable => null,
+                    int other => throw Fail(other),
+                };
+
+                static CryptographicException Fail(int result)
+                {
+                    Debug.Fail($"Unexpected result {result} from {nameof(CryptoNative_IsSignatureAlgorithmAvailable)}");
+                    return new CryptographicException();
+                }
+            }
+
+            [LibraryImport(Libraries.CryptoNative, StringMarshalling = StringMarshalling.Utf8)]
+            private static partial SafeEvpPKeyHandle? CryptoNative_MLDsaGenerateKey(string keyType, ReadOnlySpan<byte> seed, int seedLength);
+
+            public static SafeEvpPKeyHandle MLDsaGenerateKey(string algorithmName, ReadOnlySpan<byte> seed)
```

Comment:
I know we've had algorithm identifiers in sub-nested types before, but I don't think we've put functions inside them.  Most of these seem like they should be popped out a scope to Interop.Crypto